packages/perseus-editor/src/components/graph-settings.tsx

```diff
@@ -28,70 +28,111 @@ const defaultBackgroundImage = {
-        showTooltips: PropTypes.bool,
-        rulerLabel: PropTypes.string,
-        rulerTicks: PropTypes.number,
-    },
-
-    getDefaultProps: function () {
-        return {
-            editableSettings: ["graph", "snap", "image", "measure"],
-            box: [
-                interactiveSizes.defaultBoxSizeSmall,
-                interactiveSizes.defaultBoxSizeSmall,
-            ],
-            labels: ["x", "y"],
-            range: [
-                [-10, 10],
-                [-10, 10],
-            ],
-            step: [1, 1],
-            gridStep: [1, 1],
-            snapStep: [1, 1],
-            valid: true,
-            backgroundImage: defaultBackgroundImage,
-            markings: "graph",
-            showProtractor: false,
-            showRuler: false,
-            showTooltips: false,
-            rulerLabel: "",
-            rulerTicks: 10,
-        };
-    },
-
-    getInitialState: function () {
+type Props = {
+    editableSettings: any[];
+    box: readonly number[];
+    labels: readonly string[];
+    range: Coords;
+    step: readonly number[];
+    gridStep: readonly number[];
+    snapStep: number[];
+    valid: boolean;
+    backgroundImage: any;
+    markings: "graph" | "grid" | "none";
+    showProtractor?: boolean;
+    showRuler?: boolean;
+    showTooltips?: boolean;
+    rulerLabel: string;
+    rulerTicks: number;
+} & Changeable.ChangeableProps;
+
+type DefaultProps = {
+    editableSettings: Props["editableSettings"];
+    box: Props["box"];
+    labels: Props["labels"];
+    range: Props["range"];
+    step: Props["step"];
+    gridStep: Props["gridStep"];
+    snapStep: Props["snapStep"];
+    valid: Props["valid"];
+    backgroundImage: Props["backgroundImage"];
+    markings: Props["markings"];
+    rulerLabel: Props["rulerLabel"];
+    rulerTicks: Props["rulerTicks"];
+    showProtractor?: Props["showProtractor"];
+    showRuler?: Props["showRuler"];
+    showTooltips?: Props["showTooltips"];
+};
+
+type State = {
+    labelsTextbox: string[];
+    gridStepTextbox: number[];
+    snapStepTextbox: number[];
+    stepTextbox: number[];
+    rangeTextbox: any[];
+    backgroundImage: any;
+};
+
+class GraphSettings extends React.Component<Props, State> {
+    static displayName: "GraphSettings";
+
+    static defaultProps: DefaultProps = {
+        editableSettings: ["graph", "snap", "image", "measure"],
+        box: [
+            interactiveSizes.defaultBoxSizeSmall,
+            interactiveSizes.defaultBoxSizeSmall,
+        ],
+        labels: ["x", "y"],
+        range: [
+            [-10, 10],
+            [-10, 10],
+        ],
+        step: [1, 1],
+        gridStep: [1, 1],
+        snapStep: [1, 1],
+        valid: true,
+        backgroundImage: defaultBackgroundImage,
+        markings: "graph",
+        rulerLabel: "",
+        rulerTicks: 10,
+        showProtractor: false,
+        showRuler: false,
+        showTooltips: false,
+    };
+
+    _isMounted: any;
```

Comment:
I'd guess this is a `boolean`.  Can we just initialize it like this instead?

```suggestion
    _isMounted = false;
```